packages/DatePicker/src/components/Calendar/Calendar.js

```diff
@@ -35,15 +49,64 @@ function Calendar(props) {
   const I18n = useI18n();
 
   // Props
-  const { date, possibleDate, onSelect } = props;
+  const { date, isVisible, onSelect, possibleDate, resetPossibleDate } = props;
+
+  // State
+  const [shouldShowShortcut, setShouldShowShortcut] = React.useState(false);
+  const [currentMonth, setCurrentMonth] = React.useState(null);
 
   // Ref
   const nextButtonRef = React.useRef(null);
   const prevButtonRef = React.useRef(null);
+  const calendarRef = React.useRef(null);
+
+  function keepFocus() {
+    if (calendarRef.current) calendarRef.current.focus();
+  }
+
+  // Effect
+  React.useEffect(() => {
+    if (!isVisible) {
+      setShouldShowShortcut(false);
+      setCurrentMonth(null);
+    }
+    keepFocus();
+  }, [isVisible]);
```

Comment:
I think the linter should be warning you that you need to add keepFocus as part of the arrayDependency isn't?